File: BookLand_FE/src/types/EventRuleType.ts
export const EventRuleType = {
    MIN_ORDER_VALUE: 'MIN_ORDER_VALUE',
    MAX_ORDER_VALUE: 'MAX_ORDER_VALUE',
    MIN_QUANTITY: 'MIN_QUANTITY',
    MAX_QUANTITY: 'MAX_QUANTITY'
} as const;

export type EventRuleType = (typeof EventRuleType)[keyof typeof EventRuleType];

File: BookLand_BE/src/main/java/com/example/bookland_be/enums/EventRuleType.java
package com.example.bookland_be.enums;

public enum EventRuleType {
    // Giá trị đơn hàng
    MIN_ORDER_VALUE("Giá trị đơn tối thiểu (VNĐ)"),
    MAX_ORDER_VALUE("Giá trị đơn tối đa (VNĐ)"),

    // Số lượng sản phẩm
    MIN_QUANTITY("Số lượng sản phẩm tối thiểu"),
    MAX_QUANTITY("Số lượng sản phẩm tối đa");

    private final String description;

    EventRuleType(String description) {
        this.description = description;
    }

    public String getDescription() {
        return description;
    }
}